### PR TITLE
feat: Add machine and pool count to header and remove tab nav MAASENG-1341

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Added machine/pool count in header title for machine list and pool list
- Removed tab navigation from machine list, pool list, and tag list.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine list
- Ensure the number of machines/pools is displayed correctly in the machine list header
- Click on the link "(number) pools"
- Ensure you are navigated to the Pools list
- Ensure the counts are still correct
- Click on the link "(number) machines"
- Ensure you are navigated to the Machine list
- Ensure the counts are still correct

## Fixes

Fixes [MAASENG-1341](https://warthogs.atlassian.net/browse/MAASENG-1341)

### Before
image

image

image

### After
image

image

image

[MAASENG-1341]: https://warthogs.atlassian.net/browse/MAASENG-1341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by petermakowski — COMMENTED** [review]
Looks good! Have you run cypress on this? I think we may have a bunch of tests that will fail after this change and will need updating.

File: src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.tsx [hunk 1]
```diff
@@ -76,6 +68,12 @@ export const MachineListHeader = ({
     }
   }, [machinesPathMatch, selectedMachines, setSidePanelContent]);
 
+  useEffect(() => {
+    dispatch(resourcePoolActions.fetch());
+  }, [dispatch]);
+
+  const resourcePools = useSelector(resourcePoolSelectors.all);
```
Comment (petermakowski) [inline src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.tsx:75]:
```suggestion
  const resourcePoolsCount = useSelector(resourcePoolSelectors.count);
```

File: src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.tsx [hunk 2]
```diff
@@ -142,15 +140,15 @@ export const MachineListHeader = ({
...
-          selected={selectedCount}
-        />
-      }
       subtitleLoading={selectedCountLoading}
-      title={"Machines"}
+      title={
+        <>
+          {allMachineCount} machines in{" "}
+          <Link to={urls.pools.index}>
+            {resourcePools.length} {pluralize("pool", resourcePools.length)}
```
Comment (petermakowski) [inline src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.tsx:148]:
```suggestion
            {resourcePoolsCount} {pluralize("pool", resourcePoolsCount)}
```

File: src/app/pools/views/Pools.tsx [hunk 3]
```diff
@@ -10,11 +14,21 @@ import NotFound from "app/base/views/NotFound";
...
 const Pools = (): JSX.Element => {
   const base = urls.pools.index;
+  const dispatch = useDispatch();
   const { machineCount } = useFetchMachineCount();
+
+  useEffect(() => {
+    dispatch(resourcePoolActions.fetch());
+  }, [dispatch]);
+
+  const resourcePools = useSelector(resourcePoolSelectors.all);
```
Comment (petermakowski) [inline src/app/pools/views/Pools.tsx:30]: Update to use the count selector as suggested above.

**Comment by ndv99** [comment]
> Looks good! Have you run cypress on this? I think we may have a bunch of tests that will fail after this change and will need updating.

Just fixing the Cypress tests now

**Pull request merged**